android_p/google_diff/cel_apl/kernel/project-celadon/0006-Virtual-PCM-device-driver-for-enabling-BT-HFP.patch

```diff
@@ -0,0 +1,1112 @@
+Date: Thu, 7 Mar 2019 14:45:11 +0530
+Subject: [PATCH] Virtual PCM device driver for enabling BT HFP
+
+Virtual Sound card driver creates pcm sound card for
+android devices with Intel connectivity chip using USB
+transport, and will enable HFP feature
+
+Tracked-On: OAM-75510
+Signed-off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
+Signed-off-by: Laxminarayan Bharadiya, Pankaj
+               <pankaj.laxminarayan.bharadiya@intel.com>
+Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
+---
+ drivers/bluetooth/btusb.c         |    2 +
+ drivers/misc/Kconfig              |   11 +
+ drivers/misc/Makefile             |    4 +-
+ drivers/misc/btusb_sco_snd_card.c | 1030 +++++++++++++++++++++++++++++
+ 4 files changed, 1046 insertions(+), 1 deletion(-)
+ create mode 100644 drivers/misc/btusb_sco_snd_card.c
+
+diff --git a/drivers/bluetooth/btusb.c b/drivers/bluetooth/btusb.c
+index feb51f016..f5497bdfc 100644
+--- a/drivers/bluetooth/btusb.c
++++ b/drivers/bluetooth/btusb.c
+@@ -3160,10 +3160,12 @@ static int btusb_probe(struct usb_interface *intf,
+ 	if (id->driver_info & BTUSB_AMP) {
+ 		/* AMP controllers do not support SCO packets */
+ 		data->isoc = NULL;
++#if (!IS_ENABLED(CONFIG_BT_SCOHCI))
+ 	} else {
+ 		/* Interface orders are hardcoded in the specification */
+ 		data->isoc = usb_ifnum_to_if(data->udev, ifnum_base + 1);
+ 		data->isoc_ifnum = ifnum_base + 1;
++#endif
+ 	}
+ 
+ 	if (!reset)
+diff --git a/drivers/misc/Kconfig b/drivers/misc/Kconfig
+index 313494f63..c1263a31e 100644
+--- a/drivers/misc/Kconfig
++++ b/drivers/misc/Kconfig
+@@ -528,6 +528,17 @@ config UID_SYS_STATS_DEBUG
+ 	help
+ 	  Per TASK based io statistics exported to /proc/uid_io
+ 
++config BT_SCOHCI
++	tristate "HCI USB driver"
++	depends on BT
++	help
++	  Bluetooth SCO USB driver.
++	  This driver is required if you want to use Bluetooth devices with
++	  USB interface to support SCO.
++
++	  Say Y here to compile support for Bluetooth USB devices into the
++	  kernel or say M to compile it as module (btusb)
++
+ source "drivers/misc/c2port/Kconfig"
+ source "drivers/misc/eeprom/Kconfig"
+ source "drivers/misc/cb710/Kconfig"
+diff --git a/drivers/misc/Makefile b/drivers/misc/Makefile
+index 30b7c50e3..1842348ec 100644
+--- a/drivers/misc/Makefile
++++ b/drivers/misc/Makefile
+@@ -60,4 +60,6 @@ obj-$(CONFIG_OCXL)		+= ocxl/
+ obj-$(CONFIG_MISC_RTSX)		+= cardreader/
+ obj-$(CONFIG_UID_SYS_STATS)	+= uid_sys_stats.o
+ obj-$(CONFIG_INTEL_HDCP_VIRTIO) += virtio-hdcp/
+-obj-$(CONFIG_INTEL_COREU_VIRTIO)	+= virtio-coreu/
++obj-$(CONFIG_INTEL_COREU_VIRTIO) += virtio-coreu/
++obj-$(CONFIG_BT_SCOHCI)          += btusb_sco_snd_card.o
++
+diff --git a/drivers/misc/btusb_sco_snd_card.c b/drivers/misc/btusb_sco_snd_card.c
+new file mode 100644
+index 000000000..2a5706785
+--- /dev/null
++++ b/drivers/misc/btusb_sco_snd_card.c
+@@ -0,0 +1,1030 @@
++/*
++ *
++ *  Bluetooth SCO driver
++ *
++ *  Copyright (C) 2019-2020 Intel Corporation
```

Comment:
Please use license header as mentioned in below link and add all sign-off as author names.

https://elixir.bootlin.com/linux/v5.1-rc3/source/sound/soc/soc-compress.c